examples/nlp/language_modeling/prepare_truncated_data.py

```diff
@@ -54,27 +59,215 @@ def _process_line(line, task, tokenizer, max_seq_length, tokens_to_generate_extr
     truncation_seperator = line['truncation_seperator']
     truncation_seperator_tokens_len = len(tokenizer.text_to_tokens(truncation_seperator))
 
+    if remove_newline_tab:
+        question = question.replace('\n', ' ').replace('\t', ' ').strip().replace('  ', ' ')
+        question_len = len(tokenizer.text_to_tokens(question))
+
+        context = context.replace('\n', ' ').replace('\t', ' ').strip().replace('  ', ' ')
+        context_tokens = tokenizer.text_to_tokens(context)
+        context_len = len(context_tokens)
+
+        input_text = line['input']
+        input_text = input_text.replace('\n', ' ').replace('\t', ' ').strip().replace('  ', ' ')
+
     total_len = context_len + tokens_to_generate_extra + question_len
-    if total_len > max_seq_length:
-        truncated_text = (
-            tokenizer.tokens_to_text(
-                context_tokens[
-                    : (max_seq_length - tokens_to_generate_extra - truncation_seperator_tokens_len - question_len)
-                ]
+    if truncation_pos == 'right':
+        if total_len > max_seq_length:
```

Comment:
We can make `if total_len > max_seq_length:`as the **outer block** and keep the truncation blocks inside it.

This would avoid the repetitive `max_seq_length` checks inside all the truncation blocks.